### PR TITLE
Bunch potential interactives into one lambda invocation

### PR DESCRIPTION
**Pull request by NovemberTang**
## What does this change?

Previously, we were sending 10 individual messages to SNS at once. Now, we are creating a JSON list of x repositories, and submitting it in one invocation.

## Why?

Fewer lambda invocations means better performance overall. Theoretically we could send more in one go, but leaving it at 10 for now to keep us under AWS API's abuse limits, which seem to be triggered after about 10 repos have been checked. We can make the interactive monitor a bit smarter about which checks it runs in a later PR.

## How has it been verified?

Deployed to CODE and verified the messages are processed correctly

### CODE REVIEW

File: packages/interactive-monitor/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	"scripts": {
 		"build": "esbuild src/index.ts --bundle --platform=node --target=node20 --outdir=dist --external:@aws-sdk",
 		"test": "echo \"No test specified\" && exit 0",
-		"start": "APP=interactive-monitor ts-node src/run-locally.ts"
+		"start": "AWS_PROFILE=deployTools ts-node src/run-locally.ts"
```
Comment (akash1810) [inline packages/interactive-monitor/package.json:9]: Is this change intentional?

Comment (NovemberTang) [reply packages/interactive-monitor/package.json:9]: Yes. the `APP` variable isn't being used for anything, and setting the `AWS_PROFILE` means we can make S3 requests

Comment (akash1810) [reply packages/interactive-monitor/package.json:9]: If we use https://github.com/guardian/service-catalogue/blob/f3dbbdebd52e3d2185f01bbe4bdec0630a75c90e/packages/common/src/aws.ts#L22-L30 we shouldn't need to set `AWS_PROFILE`. This would also make us consistent too.

File: packages/interactive-monitor/src/index.ts [hunk 2]
```diff
@@ -100,7 +101,8 @@ async function applyTopics(repo: string, owner: string, octokit: Octokit) {
 
 export async function assessRepo(repo: string, owner: string, config: Config) {
 	const octokit = await stageAwareOctokit(config.stage);
-	const s3 = new S3Client({ region: 'us-east-1' });
+	const awsConfig = awsClientConfig(config.stage);
+	const s3 = new S3Client({ ...awsConfig, region: 'us-east-1' });
```
Comment (NovemberTang) [inline packages/interactive-monitor/src/index.ts:105]: overriding the region rather than making the region configurable, as there are unlikely to be many scenarios where eu-west-1 is not appropriate

File: packages/interactive-monitor/package.json (hunk 1 above)
Comment (NovemberTang) [reply packages/interactive-monitor/package.json]: we're using this now, see later commits

File: packages/interactive-monitor/src/index.ts [hunk 3]
```diff
@@ -134,8 +136,12 @@ export async function assessRepo(repo: string, owner: string, config: Config) {
 export const handler: SNSHandler = async (event) => {
 	const config = getConfig();
 	const owner = 'guardian';
-	const events = event.Records.map(
-		async (record) => await assessRepo(record.Sns.Message, owner, config),
+	const events = event.Records.map(async (record) =>
```
Comment (akash1810) [inline packages/interactive-monitor/src/index.ts:139]: What type is `events` here? I think it might be an array of arrays? Not sure that the `Promise.all` on line 146 would work as we'd expect, and we might want to call [`.flat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) events (or use [`flatMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)) to produce a one-dimensional array.

**Review by akash1810 — COMMENTED** [review]
added a minor comment

File: packages/interactive-monitor/src/index.ts (hunk 3 above)
Comment (NovemberTang) [reply packages/interactive-monitor/src/index.ts:139]: Having run this on CODE, all the logs look normal, what unexpected behaviour do you anticipate?

**Review by tjsilver — APPROVED** [review]
👍
